docs/Using Any API/array-response.md

```diff
@@ -0,0 +1,113 @@
+```
+
+<div class="remix-callout">
+    <a href="https://remix.ethereum.org/#url=https://docs.chain.link/samples/APIRequests/FetchFromArray.sol" target="_blank" >Open in Remix</a>
+    <a href="/docs/conceptual-overview/#what-is-remix" >What is Remix?</a>
+</div>
+
+To use this contract:
+
+1. Open the [contract in Remix](https://remix.ethereum.org/#url=https://docs.chain.link/samples/APIRequests/FetchFromArray.sol).
+
+1. Compile and deploy the contract using the Injected Web3 environment. The contract includes all the configuration variables for _Kovan_ testnet , make sure you are connected to _Kovan_. Note that in the _constructor_ , we setup:
+
+   - The Chainlink Token address for _Kovan_ by calling [`setChainlinkToken`](/docs/chainlink-framework/#setchainlinktoken) function.
+   - The Oracle contract address for _Kovan_ by calling [`setChainlinkOracle`](/docs/chainlink-framework/#setchainlinkoracle) function.
+   - The `jobId`: specific job for the oracle node to run. Note that in this case, the _id_ is a _string_ data type. Hence, we need to call a job which calls an API and returns a _string_.The job spec for the Chainlink node in this example can be [found here](/docs/direct-request-get-string/).
+
+1. Fund your contract with 0.1 LINK as shown [here](/docs/fund-your-contract/).
+
+1. Call `id` function to confirm `id` is empty.
+
+1. Run `requestFirstId` function. This builds the `Chainlink.Request` using the right parameters. Note the syntax that will be used by the oracle node to parse the _path_: `req.add("path", "0,id")` ("Fetch the id at index 0 of the array").
+
+1. After few seconds, call `id` function. You should get a non-empty response: _bitcoin_ at the time of writing.
+
+## Choosing Link token address, Oracle and JobId
+
+[`setChainlinkToken`](/docs/chainlink-framework/#setchainlinktoken) function allows to set the Link token address for the [network](/docs/link-token-contracts/) you are deploying to. [`setChainlinkOracle`](/docs/chainlink-framework/#setchainlinkoracle) function allows to set a specific Chainlink oracle that a contract makes an API call from, and `jobId` refers to a specific job for that node to run. Each job is unique and returns different types of data.
+
+For example, a job that returns a `bytes32` variable from an API would have a different `jobId` than a job that retrieved the same data, but in the form of a `uint256` variable.
+
+[market.link](https://market.link/) provides a searchable catalogue of Oracles, Jobs and their subsequent return types.
+
+## Make an Existing Job Request
+
+If your contract is calling a public API endpoint, an Oracle job may already exist for it. If so, it could mean you do not need to add the URL, or other adapter parameters into the request, since the job already configured to return the desired data. This makes your smart contract code more succinct. To see an example of a contract using an existing job, see [Make an Existing Job Request](../existing-job-request/).
```

Comment:
```suggestion
If your contract is calling a public API endpoint, an Oracle job might already exist for it. It could mean that you do not need to add the URL or other adapter parameters into the request because the job is already configured to return the desired data. This makes your smart contract code more succinct. To see an example of a contract using an existing job, see the [Make an Existing Job Request](../existing-job-request/) guide.
```